src/Fantomas/Format.fs

```diff
@@ -42,16 +62,17 @@ let private formatContentInternalAsync
                     if not isValid then
                         return InvalidCode(filename = file, formattedContent = formattedContent)
                     else
-                        return Formatted(filename = file, formattedContent = formattedContent)
+                        return
+                            Formatted(filename = file, formattedContent = formattedContent, profileInfos = profileInfos)
                 else
-                    return Unchanged(filename = file)
+                    return Unchanged(filename = file, profileInfos = profileInfos)
             with ex ->
                 return Error(file, ex)
         }
 
 let formatContentAsync = formatContentInternalAsync false
 
-let private formatFileInternalAsync (compareWithoutLineEndings: bool) (file: string) =
+let private formatFileInternalAsync (compareWithoutLineEndings: bool) (profile: bool) (file: string) =
```

Comment:
We might want to consider bundling all the parameters into a single record.
It is slowly becoming a boolean fest.

Reply:
I tried to come up with something and hope it makes somewhat sense.